### PR TITLE
Added signal_t struct and modified messages to use it.

### PR DESCRIPTION
**Pull request by ungureanuvladvictor**
\cc: @gsmcmullin @mookerji @mfine 

Signed-off-by: Vlad Ungureanu <vvu@vdev.ro>

### CODE REVIEW

**Comment by ungureanuvladvictor** [comment]
retest this please

**Comment by mfine** [comment]
Expanding SID here looks great and looks conforming to what we wanted to expand the PRN to and this helps provide more clarity around using this structure. @gsmcmullin @henryhallam any issues? Only minor nit is that the `sid` field might not be as good of a fit - maybe just `signal`?

File: python/sbp/signal.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+from sbp.utils import fmt_repr, exclude_fields, walk_json_dict, containerize, greedy_string
+
+# Automatically generated from piksi/yaml/swiftnav/sbp/signal.yaml with generate.py.
+# Please do not hand edit!
+
+
+class Signal(object):
+  """Signal.
+  
+  A struct to represent the signal(constellation, band, PRN).
```
Comment (mookerji) [inline python/sbp/signal.py]: nit: "A struct to represent" => "Signal identifier containing constellation, band, and PRN." or something.

File: spec/yaml/swiftnav/sbp/signal.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+stable: True
+public: True
+include:
+  - types.yaml
+definitions:
+
+ - Signal:
+    short_desc: Represents all the relevant information about the signal
+    desc: |
+      A struct to represent the signal(constellation, band, PRN).
```
Comment (mookerji) [inline spec/yaml/swiftnav/sbp/signal.yaml]: See nit above

**Comment by mookerji** [comment]
lgtm

**Comment by ungureanuvladvictor** [comment]
The only part I did not touch are the tests, they pass at this point but I do not know the correctness of them.

File: spec/yaml/swiftnav/sbp/signal.yaml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+package: swiftnav.sbp.signal
+description: Struct to represent the signal(constellation, band, PRN).
+stable: True
+public: True
+include:
+  - types.yaml
+definitions:
+
+ - sbp_signal:
```
Comment (mookerji) [inline spec/yaml/swiftnav/sbp/signal.yaml]: Errrr, can we keep this upper-cased?

File: spec/yaml/swiftnav/sbp/signal.yaml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    short_desc: Represents all the relevant information about the signal
+    desc: |
+     Signal identifier containing constellation, band, and PRN.
+    fields:
+        - constellation:
+            type: u8
+            desc: Constellation from which the sattelite belongs to.
```
Comment (mookerji) [inline spec/yaml/swiftnav/sbp/signal.yaml]: nit: spelling `satellite`

File: spec/yaml/swiftnav/sbp/signal.yaml (hunk 3 above)
Comment (mfine) [inline spec/yaml/swiftnav/sbp/signal.yaml]: And camel-cased.

File: spec/yaml/swiftnav/sbp/signal.yaml [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+#
+# This source is subject to the license found in the file 'LICENSE' which must
+# be be distributed together with this source. All other rights reserved.
+#
+# THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+# EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+
+package: swiftnav.sbp.signal
+description: Struct to represent the signal(constellation, band, PRN).
```
Comment (fnoble) [inline spec/yaml/swiftnav/sbp/signal.yaml]: Space after "signal"

File: spec/yaml/swiftnav/sbp/signal.yaml (hunk 3 above)
Comment (fnoble) [inline spec/yaml/swiftnav/sbp/signal.yaml]: As this is not a message, I think it should be CamelCased for consistency, i.e. `SBPSignal`

File: spec/yaml/swiftnav/sbp/acquisition.yaml [hunk 6]
```diff
@@ -40,12 +41,9 @@ definitions:
...
             desc: Carrier frequency of best point
         - sid:
-            type: u32
+            type: sbp_signal
             desc: |
-              Signal identifier of the satellite signal for which acquisition
-              was attempted - values 0x00 through 0x1F represent GPS PRNs 1
-              through 32 respectively (PRN-1 notation); other values reserved
-              for future use.
+              Signal identifier of the satellite.
```
Comment (fnoble) [inline spec/yaml/swiftnav/sbp/acquisition.yaml]: I think adding "of the satellite" is not quite correct. I'd suggest "Signal identifier (constellation / satellite / band)".

File: spec/yaml/swiftnav/sbp/observation.yaml [hunk 7]
```diff
@@ -86,11 +87,9 @@ definitions:
               signal has lost and regained lock, indicating that the
               carrier phase ambiguity may have changed.
         - sid:
-            type: u32
+            type: sbp_signal
             desc: |
-              Signal identifier of the satellite signal - values 0x00
-              through 0x1F represent GPS PRNs 1 through 32 respectively
-              (PRN-1 notation); other values reserved for future use.
+              Signal identifier of the satellite signal.
```
Comment (fnoble) [inline spec/yaml/swiftnav/sbp/observation.yaml]: Again I'd suggest "Signal identifier (constellation / satellite / band)".

File: spec/yaml/swiftnav/sbp/observation.yaml [hunk 8]
```diff
@@ -244,11 +243,9 @@ definitions:
           type: u8
           desc: Satellite is healthy?
       - sid:
-          type: u32
+          type: sbp_signal
           desc: |
-            Signal identifier being tracked - values 0x00 through 0x1F represent
-            GPS PRNs 1 through 32 respectively (PRN-1 notation); other values
-            reserved for future use
+            Signal identifier of the satellite beign tracked.
```
Comment (fnoble) [inline spec/yaml/swiftnav/sbp/observation.yaml]: I'd suggest "Identifier of the signal being tracked (constellation / satellite / band)".

File: spec/yaml/swiftnav/sbp/signal.yaml [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+definitions:
+
+ - sbp_signal:
+    short_desc: Represents all the relevant information about the signal
+    desc: |
+     Signal identifier containing constellation, band, and PRN.
+    fields:
+        - constellation:
+            type: u8
+            desc: Constellation from which the satellite belongs to.
```
Comment (fnoble) [inline spec/yaml/swiftnav/sbp/signal.yaml]: "Constellation to which the satellite belongs"

File: spec/yaml/swiftnav/sbp/signal.yaml [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+        - band:
+            type: u8
+            desc: Band of the satellite.
```
Comment (fnoble) [inline spec/yaml/swiftnav/sbp/signal.yaml]: "Signal band"

File: spec/yaml/swiftnav/sbp/signal.yaml [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+        - prn:
+            type: u16
+            desc: PRN of the satellite.
```
Comment (fnoble) [inline spec/yaml/swiftnav/sbp/signal.yaml]: "Signal PRN"

File: spec/yaml/swiftnav/sbp/tracking.yaml [hunk 12]
```diff
@@ -35,11 +36,9 @@ definitions:
                         - 0: Disabled
                         - 1: Running
         - sid:
-            type: u32
+            type: sbp_signal
             desc: |
-              Signal identifier being tracked - values 0x00 through 0x1F
-              represent GPS PRNs 1 through 32 respectively (PRN-1 notation);
-              other values reserved for future use
+              Signal identifier being tracked.
```
Comment (fnoble) [inline spec/yaml/swiftnav/sbp/tracking.yaml]: Identifier of signal being tracked

File: spec/yaml/swiftnav/sbp/tracking.yaml [hunk 13]
```diff
@@ -81,11 +80,9 @@ definitions:
           type: u8
           desc: Tracking channel of origin
       - sid:
-          type: u32
+          type: sbp_signal
           desc: |
-            Signal identifier being tracked - values 0x00 through 0x1F
-            represent GPS PRNs 1 through 32 respectively (PRN-1 notation);
-            other values reserved for future use
+            Signal identifier being tracked.
```
Comment (fnoble) [inline spec/yaml/swiftnav/sbp/tracking.yaml]: Identifier of signal being tracked

**Comment by fnoble** [comment]
Just some documentation wording fixes but lgtm

**Comment by ungureanuvladvictor** [comment]
Fixed all the PR comments

**Comment by ungureanuvladvictor** [comment]
Fixed all the PR comments

**Comment by gsmcmullin** [comment]
Rebased in  #281

**Pull request closed without merging**